### ISSUE TITLE
Dataset preparation

### ISSUE THREAD

**Issue by mrengineer** [srihari-humbarwadi/DeepLabV3_Plus-Tensorflow2.0#12]
Hi. While train you are using labelIds pngs from cityscapes. There are gray masks. Please, explain how code make difference between classes.

+ There is cityscapes_dict.pkl for inference. How was it made? Why pkl instead array?

**Comment by srihari-humbarwadi** [comment]
-  In the grayscale masks, each pixel has a value that corresponds to the class_id.
 - There is no need to one_hot encode your target masks if you are using `sparse_categorical_crossentropy` as the loss function.
 - cityscapes_dict.pkl has a dict which maps class_id to RGB color values. This is useful when you want to visualize the model output.

**Comment by mrengineer** [comment]
thank you for clarify

**Comment by Alokksinha00** [comment]
> Hi. While train you are using labelIds pngs from cityscapes. There are gray masks. Please, explain how code make difference between classes.
> 
> * There is cityscapes_dict.pkl for inference. How was it made? Why pkl instead array?

Hi @mrengineer, could you please tell me which exact dataset you took for inference and training?
if you want to run without training first, which all places in inference and train python file have to be modified?
